### PR TITLE
Update a bunch of URLs.

### PR DESCRIPTION
**Pull request by Carreau**
1) https-ify.
2) move all scipy.github.io/devdocs to docs.scipy.org/doc/scipy
3) code.entought does not exists anymore
4) Siam redirect to archive.siam, update accordingly.

Note that content/en/press-kit.md contain:

> the SciPy logo [here](https://github.com/scipy/scipy/tree/main/branding/logo)

But this seem to have been taken from numpy ad I can't find a trace,
even in history of the branding folder.

### CODE REVIEW

**Comment by Carreau** [comment]
Note that I'm partially unsure about the move of devdocs, and maybe this should be reviewed on a case-by-case basis ?

File: config.yaml [hunk 1]
```diff
@@ -97,7 +97,7 @@ languages:
             - text: Citing SciPy
               link: /citing-scipy
             - text: Roadmap
-              link: https://scipy.github.io/devdocs/dev/roadmap.html
+              link: https://docs.scipy.org/doc/scipy/dev/roadmap.html
```
Comment (tupui) [inline config.yaml:100]: This one could stay with dev docs as it should be updated as we go and it should be visible.

File: content/en/contribute.md [hunk 2]
```diff
@@ -24,13 +24,13 @@ If you are new to contributing to open source, we also highly recommend reading
...
 We have a
-[Code of Conduct](http://scipy.github.io/devdocs/dev/conduct/code_of_conduct.html)
+[Code of Conduct](https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html)
 to foster an open and welcoming environment.
 
 ### Writing code
 
 Programmers, this
-[guide](https://scipy.github.io/devdocs/dev/contributor/development_workflow.html#development-workflow)
+[guide](https://docs.scipy.org/doc/scipy/dev/contributor/development_workflow.html#development-workflow)
```
Comment (tupui) [inline content/en/contribute.md:33]: Here too, dev docs is better as we always want to point to latest for such instructions.

File: content/en/download.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ To check out the latest **SciPy** sources:
 
 Build instructions for SciPy can be found in its documentation. The
 latest version can be found at:
-<https://scipy.github.io/devdocs/dev/contributor/building.html>
+<https://docs.scipy.org/doc/scipy/dev/contributor/building.html>
```
Comment (tupui) [inline content/en/download.md:50]: Same.

File: content/en/install.md [hunk 4]
```diff
@@ -100,4 +100,4 @@ macOS doesn't have a preinstalled package manager, but you can install
 A word of warning: building SciPy from source can be a nontrivial exercise. We
 recommend using binaries instead if those are available for your platform.
 For details on how to build from source, see
-[this guide in the SciPy docs](http://scipy.github.io/devdocs/dev/contributor/building.html).
+[this guide in the SciPy docs](https://docs.scipy.org/doc/scipy/dev/contributor/building.html).
```
Comment (tupui) [inline content/en/install.md:103]: Same.

**Review by tupui — CHANGES_REQUESTED** [review]
Thanks @Carreau, as you pointed out there are a few occurrences of dev docs that should stay in. The rest LGTM.

**Comment by Carreau** [comment]
There are also a number of links in the scipy source itself, in particular README.rst link to the CoC in `scipy.github.io`, should I send a PR for that as well ?

**Review by tupui — APPROVED** [review]
Thanks for the update. LGTM

**Comment by tupui** [comment]
> There are also a number of links in the scipy source itself, in particular README.rst link to the CoC in `scipy.github.io`, should I send a PR for that as well ?

Sure, just not high priority unless the links are broken.

**Pull request merged**